cms/core/models/snippets.py

```diff
@@ -88,16 +88,7 @@ class GlossaryTerm(TranslatableMixin, PreviewableMixin, RevisionMixin, index.Ind
     ]
 
     class Meta:
-        constraints: ClassVar[list[models.BaseConstraint]] = [
-            models.UniqueConstraint(
-                Lower("name"),
-                name="core_glossary_term_name_unique",
-                violation_error_message="A glossary term with this name already exists.",
-            ),
-            models.UniqueConstraint(
-                fields=("translation_key", "locale"), name="unique_translation_key_locale_core_glossaryterm"
-            ),
-        ]
+        unique_together: ClassVar[list[tuple[str, ...]]] = [*TranslatableMixin.Meta.unique_together, ("name", "locale")]
```

Comment:
The original solution didn't work. It seems that it only works with `unique_together`.

References:
https://github.com/wagtail/wagtail/issues/8918#issuecomment-1208670360 
https://enzedonline.com/en/tech-blog/dealing-with-unique-fields-on-a-multi-lingual-site/